src/agent/state.js

```diff
@@ -361,17 +361,29 @@ StateResolver.prototype.resolveLocalsList_ = function (frame, args) {
   var self = this;
   var usedNames = {};
   var makeMirror = this.ctx_.MakeMirror;
-  return flatten(frame.allScopes().map(
+  const allScopes = frame.allScopes();
+  const count = allScopes.length;
+
+  // There will always be at least 3 scopes.
+  // For top-level breakpoints: [local, script, global]
+  // Other: [..., closure (module IIFE), script, global]
+  assert(count >= 3);
```

Comment:
It would be great if these could be `DCHECK`s. There are probably other sanity `DECHECK`s we could benefit from in this file.